dynsampler.go

```diff
@@ -13,4 +13,12 @@ type Sampler interface {
 	// should call it with whatever key you choose to use to partition traffic
 	// into different sample rates.
 	GetSampleRate(string) int
+
+	// SaveState returns a byte array containing the state of the Sampler implementation.
+	// It can be used to persist state between process restarts.
+	SaveState() ([]byte, error)
+
+	// LoadState accepts a byte array containing the serialized, previous state of the sampler
+	// implementation. It should be called before `Start`.
+	LoadState([]byte) error
```

Comment:
Adding these will need a bump in the library version because once this lands, existing objects that implement the `Sampler` interface won't anymore.